server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -303,6 +304,12 @@ protected void search(List<LeafReaderContext> leaves, Weight weight, Collector c
      * the provided <code>ctx</code>.
      */
     private void searchLeaf(LeafReaderContext ctx, Weight weight, Collector collector) throws IOException {
+
+        // Check if at all we need to call this leaf for collecting results.
+        if (!canMatch(ctx)) {
```

Comment:
nitpick: OpenSearch follows `canMatch(ctx) == false` for negative assertions.

Reply:
Done, thanks !